### PR TITLE
Use ServeTLS to call golang's private setupHTTP2_Serve method to setup HTTP2 advertisement properly

### PR DESCRIPTION
**Pull request by SuperSandro2000**
Tested this without any config changes and now my browser uses HTTP2 like it should. The certificate loading which was done by hand before, is now done all by go, so we fix a bug and save LOC.

### CODE REVIEW

**Review by antifuchs — APPROVED** [review]
For a moment I was extremely confused about the changes to the conditional... but that was just my own inability to tell `==` and `!=` apart sometimes. This looks great!

**Comment by antifuchs** [comment]
Oooooh, so the e2e tests DID immediately catch something (and here I thought they were buggy or the timeouts were too short)! Check it out: https://github.com/boinkor-net/tsnsrv/actions/runs/8084903139/job/22091260407#step:5:3552

The e2e test configures tsnsrv to spawn a plaintext listener; but tsnet says `tsnsrv-basic-start[1103]: 2024/02/28 17:59:42 could not listen: listener for &{/var/lib/tsnsrv-basic/tsnet-tsnsrv FileStore("/var/lib/tsnsrv-basic/tsnet-tsnsrv/tailscaled.state") basic 0x5033e0 false 75519ccce6351b5e787a4c68b2ab702f2884b2cb2098f9ce http://127.0.0.1:8080/ false 0 <nil> {{{} 1} {0 0}} <nil> 0xc0001cad88 0xc0001ec180 0xc0002ef520 /var/lib/tsnsrv-basic/tsnet-tsnsrv basic 0xc0002e7270 0x4a0c80   <nil> 0xc000114c40 0xc000114c80 0xc000172000 0xc0002b6608 0xc0000e30e0 [79 3 132 137 239 57 108 70 148 174 224 152 27 80 142 69 107 90 34 63 196 86 167 87 171 101 104 88 76 199 25 111] {0 0} map[] map[] 0xc0002ef790 false}: tsnet: you must enable HTTPS in the admin panel to proceed. See https://tailscale.com/s/https` - sounds like it's listening with TLS turned on regardless. I'll try to step through the logic again and see what's up.

**Comment by SuperSandro2000** [comment]
Should be fixed now

File: cli.go [hunk 1]
```diff
@@ -293,23 +298,7 @@ func (s *TailnetSrv) listen(srv *tsnet.Server) (net.Listener, error) {
...
-		if err != nil {
-			return nil, fmt.Errorf("listener for %v: %w", srv, err)
-		}
-
-		l = tls.NewListener(ln, &tls.Config{
-			Certificates: []tls.Certificate{cer},
-			MinVersion:   tls.VersionTLS12,
-		})
-	case !s.ServePlaintext:
+	case !s.ServePlaintext && (s.certificateFile == "" || s.keyFile == ""):
```
Comment (antifuchs) [inline cli.go:301]: Oh man, Boolean operator precedence gets me every time. Every time. Thanks!!

**Pull request merged**